File: server/src/entities/Upvote.ts
import { BaseEntity, Column, Entity, ManyToOne, PrimaryColumn } from 'typeorm';
import { Post } from './Post';
import { User } from './User';

@Entity()
export class Upvote extends BaseEntity {
  @Column({ type: 'int' })
  value: number;

  @PrimaryColumn()
  userId: number;

  @ManyToOne(() => User, (user) => user.upvotes, { onDelete: 'CASCADE' })
  user: User;

  @PrimaryColumn()
  postId: number;

  @ManyToOne(() => Post, (post) => post.upvotes, {
    onDelete: 'CASCADE',
  })
  post: Post;
}

File: server/models/upvote.go
package models

type Upvote struct {
	Value  int  `json:"value"`
	UserID int  `json:"userId" gorm:"primary_key, constraint: OnDelete: CASCADE"`
	User   User `json:"user"`
	PostID int  `json:"postId" gorm:"primary_key, constraint: OnDelete: CASCADE"`
	Post   Post `json:"post"`
}
